zeebe/qa/update-tests/src/test/java/io/camunda/zeebe/test/VersionCompatibilityMatrix.java

```diff
@@ -200,6 +242,44 @@ public Stream<SemanticVersion> discoverVersions() {
     return versions.stream().filter(version -> !unreleasedVersions.contains(version));
   }
 
+  private static SemanticVersion parseVersion(final String version) {
+    return SemanticVersion.parse(version)
+        .orElseThrow(
+            () -> new IllegalArgumentException("Invalid semantic version string: " + version));
+  }
+
+  private static boolean isKnownIncompatibleUpgrade(
```

Comment:
🔧 I think we can just define method `isCompatible`, not sure why but I found it a bit more confusing to read, but it's just personal taste